scripts/check-gmp.gmp

```diff
@@ -0,0 +1,1549 @@
+        self.cursor = None
+        self.con_db = None
+        self.path = path
+        self.pid = os.getpid()
+
+        # Try to read file with informations about cached reports
+        # First check whether the file exist or not
+        exist = os.path.isfile(path)
+        logger.debug("DB file exist?: %s " % exist)
+
+        if not exist:
+            os.makedirs(os.path.dirname(path), exist_ok=True)
+
+            # Connect to db
+            self.connect_db()
+
+            # Create the tables
+            self.cursor.execute(
+                """CREATE TABLE Report(
+                host text,
+                scan_end text,
+                params_used text,
+                report text
+            )"""
+            )
+
+            self.cursor.execute(
+                """CREATE TABLE Instance(
+                created_at text,
+                pid integer,
+                pending integer default 0
+            )"""
+            )
+
+            logger.debug("Tables created")
+        else:
+            self.connect_db()
+
+    @staticmethod
+    def _to_sql_bool(pending):
+        """ Replace True/False with 1/0.
+        """
+        return '1' if pending else '0'
+
+    def connect_db(self):
+        """Connect to the databse
+
+        Simply connect to the database at location <path>
+        """
+        try:
+            logger.debug("connect db: %s" % self.path)
+            self.con_db = sqlite3.connect(self.path)
+            self.cursor = self.con_db.cursor()
+            logger.debug(sqlite3.sqlite_version)
+        except Exception as e:
+            logger.debug(e)
+
+    def close_db(self):
+        """Close database
+        """
+        self.con_db.close()
+
+    def set_host(self, host):
+        """Sets the host variable
+
+        Arguments:
+            host (string): Given ip or hostname of target.
+        """
+        self.host = host
+
+    def is_old_report(self, last_scan_end, params_used):
+        """Decide whether the current report is old or not
+
+        At first the last scanend and the params that were used are fetched
+        from the database. If no report is fetched, then True will be returned.
+        The next step is to compare the old and the new scanend.
+        If the scanends matches, then return False, because it is the same
+        report. Else the old report will be deleted.
+
+        Arguments:
+            last_scan_end (string): Last scan end of report
+            params_used (string): Params used for this check
+
+        Returns:
+            True if it is an old report or empty. False if it is the same report.
+        """
+
+        # Before we do anything here, check existing instance
+
+        # Retrieve the scan_end value
+        self.cursor.execute(
+            "SELECT scan_end, params_used FROM Report WHERE" " host=?",
+            (self.host,),
+        )
+        db_entry = self.cursor.fetchone()
+
+        logger.debug("%s %s" % (db_entry, last_scan_end))
```

Comment:
Oh i suppose it should be
```suggestion
        logger.debug("%s %s", db_entry, last_scan_end)
```

See https://docs.python.org/3/howto/logging.html#logging-variable-data

This results in a small performance improvement, by only creating the output string if debug log is enabled. With the current code the log string will always be created but maybe unused if log is not enabled.